source/jormungandr/jormungandr/instance.py

```diff
@@ -745,21 +745,21 @@ def _reap_sockets(connector):
 
     @contextmanager
     def socket(self, context):
-
-        socket = None
         sockets = self._sockets
         socket_path = self.socket_path
 
+        t = None
         try:
-            socket, _ = sockets.pop()
+            socket, t = sockets.pop()
         except IndexError:  # there is no socket available: lets create one
             socket = context.socket(zmq.REQ)
             socket.connect(socket_path)
+
         try:
             yield socket
         finally:
             if not socket.closed:
```

Comment:
Maybe we should also test here if the TTL is expired, and close the socket if this is the case ?
Otherwise we may put back in the pool a socket with an expired TTL, which will be used before the reaper close it

Reply:
I agree with you, that's what has been done in asgard... I looked into the codes, there will be some work to do.. and I don't want to complicate this PR too much...

Reply:
That is not just adding here somthing like 
```
if time.now() > t:
   socket.close()
```
?